tests/Renderer/QueryResultsRenderer.test.ts

```diff
@@ -117,3 +118,90 @@ describe('QueryResultsRenderer - responding to file edits', () => {
         expect(renderer.query.explainQuery()).toContain('path includes newPath.md');
     });
 });
+
+describe('QueryResultsRenderer - internal heading links', () => {
+    let allTasks: Task[];
+
+    beforeAll(() => {
+        allTasks = readTasksFromSimulatedFile(internal_heading_links_test);
+    });
+
+    async function testHeadingLinks(taskDescription: string, queryFilePath: string, expectedLinks: string[]) {
+        const task = allTasks.find((t) => t.description.includes(taskDescription));
+        if (!task) throw new Error(`Task with description "${taskDescription}" not found`);
+
+        const renderer = makeQueryResultsRenderer('', new TasksFile(queryFilePath));
+        const queryRendererParameters = {
+            allTasks: [task],
+            allMarkdownFiles: [],
+            backlinksClickHandler: () => Promise.resolve(),
+            backlinksMousedownHandler: () => Promise.resolve(),
+            editTaskPencilClickHandler: () => Promise.resolve(),
+        };
+        const container = document.createElement('div');
+
+        await renderer.render(State.Warm, [task], container, queryRendererParameters);
+
+        const renderedText = container.innerHTML;
+        expectedLinks.forEach((link) => expect(renderedText).toContain(link));
+    }
+
+    it('should not modify internal heading links when rendering in same file', async () => {
+        await testHeadingLinks('[[#Basic Internal Links]]', 'original.md', [
+            '[[Test Data/internal_heading_links.md#Basic Internal Links|Basic Internal Links]]',
```

Comment:
I am a bit confused by this.
I _think_ the test is showing that the link _has_ been changed, which I would expect because:

1. The file names are different
2. Code coverage shows that the expression that says 'paths are the same' is never true.

It would be fine to fix the test so that it really does reach the code that the tests the path.

Having said that, the test should _ideally_ be that the same Task object is returned by the function, and not whether or not the links have been changed.

Reply:
I changed this test to look like this, passing in the path to the task as the path to the query:

```ts
        await testHeadingLinks('[[#Basic Internal Links]]', 'Test Data/internal_heading_links.md', [
            '[[Test Data/internal_heading_links.md#Basic Internal Links|Basic Internal Links]]',
        ]);
```

The test failure then shows this output, which it took me a while to understand:

![image](https://github.com/user-attachments/assets/f97fe008-b0fc-4060-b980-ee464bfd4c72)

When I changed the test to this, updating the search string, it passed - and I think is now doing what was intended.

```ts
        await testHeadingLinks('[[#Basic Internal Links]]', 'Test Data/internal_heading_links.md', [
            '[[#Basic Internal Links]]',
        ]);
```

Reply:
I updated the tests to rely on inline snapshots instead, which I think will simplify the testing logic on our end and also keep them readable

Reply:
The inline snapshots are great. And I see you have fixed the "same filename"-type test. Thanks!

Resolving conversation.